src/main/java/edu/cornell/kfs/module/cg/businessobject/lookup/CuAwardLookupableHelperServiceImpl.java

```diff
@@ -0,0 +1,131 @@
+    private static final long serialVersionUID = 1372707190594999024L;
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(CuAwardLookupableHelperServiceImpl.class);
+    
+    private static final String VIEW_INVOICES_LINK_VALUE = "View Invoices";
+    private static final String INVOICE_LINK_COLUMN_NAME = "invoiceLink";
+    
+    protected DocumentHelperService documentHelperService;
+    protected PermissionService permissionService;
+    
+    @Override
+    public List<HtmlData> getCustomActionUrls(BusinessObject businessObject, List pkNames) {
+        LOG.debug("getCustomActionUrls, entering");
+        List<HtmlData> anchorHtmlDataList = new ArrayList<HtmlData>();
+        
+        if (canInitAward()) {
+            anchorHtmlDataList.add(getUrlData(businessObject, KRADConstants.MAINTENANCE_EDIT_METHOD_TO_CALL, pkNames));
+            if (allowsMaintenanceNewOrCopyAction()) {
+                anchorHtmlDataList.add(getUrlData(businessObject, KRADConstants.MAINTENANCE_COPY_METHOD_TO_CALL, pkNames));
+            }
+        }
+
+        if (canViewInvoiceLink() && getAccountsReceivableModuleBillingService().isContractsGrantsBillingEnhancementActive()) {
+            AnchorHtmlData invoiceUrl = getInvoicesLookupUrl(businessObject);
+            anchorHtmlDataList.add(invoiceUrl);
+        }
+
+        return anchorHtmlDataList;
+    }
+    
+    @Override
+    public List<Column> getColumns() {
+        List<Column> columns = super.getColumns();
+
+        if (canInitAward() || !canViewInvoiceLink()) {
+            LOG.debug("getColumns, remove invoices column");
+            for (Iterator<Column> it = columns.iterator(); it.hasNext(); ) {
+                Column column = it.next();
+                if (StringUtils.equalsAnyIgnoreCase(column.getPropertyName(), INVOICE_LINK_COLUMN_NAME)) {
+                    it.remove();
+                }
+            }
+        }
+
+        return columns;
+    }
+    
+    @Override
+    public Collection<? extends BusinessObject> performLookup(LookupForm lookupForm, Collection<ResultRow> resultTable, boolean bounded) {
+        LOG.debug("performLookup, entering");
+        Collection<? extends BusinessObject> results = super.performLookup(lookupForm, resultTable, bounded);
+        if (canViewInvoiceLink() && !canInitAward()) {
+            resultTable.stream().forEach(this::findAndResetInvoiceColumn);
+        }
+        return results;
+    }
+    
+    protected void findAndResetInvoiceColumn(ResultRow row) {
+        row.getColumns().stream()
+            .filter(col -> StringUtils.equalsAnyIgnoreCase(col.getPropertyName(), INVOICE_LINK_COLUMN_NAME))
+            .findFirst()
+            .ifPresent(col -> resetInvoiceColumn(row, col));
+    }
+
+    protected void resetInvoiceColumn(ResultRow row, Column invoiceColumn) {
+        String link = parseHrefLinkFromAnchorTag(row.getActionUrls());
```

Comment:
The action URLs might be empty at this point if the CGB enhancement is turned off and/or the user doesn't have the right permissions. Should a check be added to make sure the parsed link is non-blank?